tests/late_joiners.py

```diff
@@ -156,111 +130,79 @@ def run(args):
-            wait_for_late_joiner(first_node, late_joiner)
-
-            if not args.skip_suspension:
-                # kill the old node(s) and ensure we are still making progress with the new one(s)
-                for node in nodes_to_kill:
-                    LOG.info(f"Stopping node {node.node_id}")
-                    node.stop()
-
-                wait_for_nodes(nodes_to_keep, catchup_msg, 3000)
-
-                cur_primary, _ = network.find_primary()
-                cur_primary_id = cur_primary.node_id
-
-                # first timer determines after how many seconds each node will be suspended
-                timeouts = []
-                suspended_nodes = []
-                for i, node in enumerate(nodes_to_keep):
-                    # if pbft suspend half of them including the primary
-                    if i % 2 != 0 and args.consensus == "pbft":
-                        continue
-                    LOG.success(f"Will suspend node with id {node.node_id}")
-                    t = random.uniform(0, 2)
-                    LOG.info(f"Initial timer for node {node.node_id} is {t} seconds...")
-                    timeouts.append((t, node))
-                    suspended_nodes.append(node.node_id)
-
-                for t, node in timeouts:
-                    suspend_time = election_timeout
-                    if node.node_id == cur_primary_id and args.consensus == "pbft":
-                        # if pbft suspend the primary for more than twice the election timeout
-                        # in order to make sure view changes will be triggered
-                        suspend_time = 2.5 * suspend_time
-                    tm = Timer(t, timeout_handler, args=[node, True, suspend_time])
-                    tm.start()
-
-                run_requests(
-                    nodes_to_keep,
-                    2 * TOTAL_REQUESTS,
-                    2001,
-                    final_msg,
-                    4000,
-                    cant_fail=False,
-                )
-
-                term_info.update(find_primary(network))
-
-                wait_for_nodes(nodes_to_keep, final_msg, 5000)
-
-                # we have asserted that all nodes are caught up
-                # assert that view changes actually did occur
-                assert len(term_info) > 1
-
-                LOG.success("----------- terms and primaries recorded -----------")
-                for term, primary in term_info.items():
-                    LOG.success(f"term {term} - primary {primary}")
+        # kill the old node(s) and ensure we are still making progress with the new one(s)
+        for backup_to_retire in nodes_to_kill:
+            LOG.success(f"Stopping node {backup_to_retire.node_id}")
+            backup_to_retire.stop()
+
+        # check nodes are ok after we killed one off
+        test_run_txs(
+            network=network,
+            args=args,
+            nodes=nodes_to_keep,
+            num_txs=len(nodes_to_keep),
```

Comment:
Not clear why sending as many transactions as there are nodes?

Reply:
one for each node